influence_benchmark/RL/base_iteration.py

```diff
@@ -146,39 +144,54 @@ def _save_kwargs(self, kwargs):
         with open(str(self.traj_dir / "kwargs.yaml"), "w+") as outfile:
             yaml.dump(self.kwargs_to_save, outfile, default_flow_style=False)
 
+    def setup_backends(self, agent_device, env_device, lora_path=None):
+        backends = {}
+        if agent_device != env_device or self.inference_quantization is not None:
+            # Assuming that if this is the case, we can't share any backend at all. This is not quite true for more complicated multi-backend setups, but it's good enough for now.
+            for model_type, model_name in self.model_names.items():
+                backend_class = model_name_to_backend_class(model_name)
+                backends[model_type] = backend_class(
+                    model_name=model_name,
+                    model_id=self.agent_model_id,  # type: ignore
+                    device=agent_device if model_type == "agent" else env_device,
```

Comment:
This is assuming we want the PM TM ID and character models on the same device (unless using gpt) which is probably true but maybe slightly reduces the point of making such a general refactor?

Reply:
in theory we could have one device for each, but maybe that is just unnecessarily complicated

Reply:
Yeah I agree that the backend refactor is a bit too general, but i think it would be useful groundwork for if we want to make stuff what you mention work for the future